taskipy/task_runner.py

```diff
@@ -28,40 +29,43 @@ def list(self):
         formatter = HelpFormatter(self.__project.tasks.values())
         formatter.print()
 
-    def run(self, task_name: str, args: List[str]) -> int:
+    def run(self, task_name: str, task_args: List[str]) -> int:
         try:
-            task = self.__project.tasks[task_name]
+            main_task = self.__project.tasks[task_name]
         except KeyError:
             raise TaskNotFoundError(task_name)
 
-        pre_task = self.__pre_task(task_name)
-        if pre_task is not None:
-            exit_code = self.__run_command_and_return_exit_code(pre_task)
-            if exit_code != 0:
-                return exit_code
+        tasks: List[Tuple[Optional[Task], List[str]]] = [
+            (self.__pre_task(task_name), []),
+            (main_task, task_args),
+            (self.__post_task(task_name), []),
+        ]
+        for task, args in tasks:
+            if task is None:
+                continue
 
-        exit_code = self.__run_command_and_return_exit_code(task, args)
-        if exit_code != 0:
-            return exit_code
-
-        post_task = self.__post_task(task_name)
-        if post_task is not None:
-            exit_code = self.__run_command_and_return_exit_code(post_task)
+            exit_code = self.__execute_command(task, args)
             if exit_code != 0:
                 return exit_code
 
         return 0
 
-    def __run_command_and_return_exit_code(self, task: Task, args: List[str] = None) -> int:
+    def __pre_task(self, task_name: str) -> Optional[Task]:
+        return self.__project.tasks.get(f'pre_{task_name}')
+
+    def __post_task(self, task_name: str) -> Optional[Task]:
+        return self.__project.tasks.get(f'post_{task_name}')
+
+    def __execute_command(self, task: Task, args: Optional[List[str]] = None) -> int:
```

Comment:
this too, I think that the older method name was self explanatory and did not need refactoring :)